sites/all/modules/custom/jdt_user_notes/jdt_user_notes.module

```diff
@@ -5,25 +5,380 @@
+    $note_container = 'note-container-' . $note['note_id'];
+    $output[$note_container] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('note-container'),
+      ),
+    );
+
+    $output[$note_container]['note-text'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('note-text'),
+      ),
+    );
+
+    $output[$note_container]['note-text']['text'] = array(
+      '#type' => 'markup',
+      '#markup' => $note['note_text'],
+    );
+
+    $output[$note_container]['node-title'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('node-title'),
+      ),
+    );
+
+    $output[$note_container]['node-title']['title'] = array(
+      '#type' => 'markup',
+      '#markup' => $article->title,
+    );
+
+    $output[$note_container]['node-author'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('node-author'),
+      ),
+    );
+
+    $output[$note_container]['node-author']['author'] = array(
+      '#type' => 'markup',
+      '#markup' => $user->name,
+    );
+
+    $output[$note_container]['edit-link'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => 'edit-link',
+      ),
+    );
+
+    $link = 'node/' . $note['article_nid'] . '/edit/' . $note['note_id'] . '/nojs';
+    $output[$note_container]['edit-link']['link'] = array(
+      '#type' => 'markup',
+      '#markup' => l(t('Edit Note'), $link, array(
+        'attributes' => array(
+          'class' => 'ctools-use-modal',
+        ),
+      )),
+    );
+
+  }
+
+  return $output;
+}
+
+/**
+ * Validates both the add and edit forms.
+ *
+ * Ensures note_title and note_body do not exceed 255 characters.
+ */
+function jdt_user_notes_validate($form, &$form_state) {
+  if (strlen($form_state['values']['note_title']) > 254) {
+    form_set_error('note_title', t('Your note title must be less than 255 
+      characters.'));
+  }
+  if (strlen($form_state['values']['note_body']) > 254) {
+    form_set_error('note_body', t('Your note body must be less than 255 
+      characters.'));
+  }
+}
+
+/**
+ * Access callback function for notes page.
+ */
+function jdt_user_notes_access($article, $permission_type, $note_id = NULL) {
+  global $user;
+
+  if (!user_access($permission_type)) {
+    return FALSE;
+  }
+
+  switch ($permission_type) {
+    case 'create new note' :
+      return TRUE;
+
+    case 'edit own note' :
+      $note = jdt_user_notes_read($note_id);
+
+      return ($note['author_uid'] == $user->uid);
+
+    case 'read own notes' :
+      $article_notes = jdt_user_notes_read_article($article->nid, $user->uid);
+
+      // Check if any notes exist, and if any belong to user.
+      $my_notes = 0;
+      foreach ($article_notes as $note) {
+        if ($note['author_uid'] == $user->uid) {
+          $my_notes++;
+        }
+      }
```

Comment:
Suggestion: since we don't care how many notes it is, plus the `jdt_user_notes_read_article()` should only returns notes of this author anyway, we can just check for `!empty($article_notes)`